File: generate-icons.py
#!/usr/bin/env python3
# Generate amber circle PNG icons at 16, 32, 48, 128 sizes
import os
from PIL import Image, ImageDraw

ICONS_DIR = os.path.expanduser("~/tabamber/icons")
SIZES = [16, 32, 48, 128]
COLORS = ["#d4a017", "#f0c040", "#b8920f"]  # outer, inner, core

for size in SIZES:
    img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    cx, cy = size // 2, size // 2
    r = int(size * 0.42)

    # Outer ring
    draw.ellipse([cx - r, cy - r, cx + r, cy + r], fill=COLORS[0])
    # Inner glow
    r2 = int(r * 0.7)
    draw.ellipse([cx - r2, cy - r2, cx + r2, cy + r2], fill=COLORS[1])
    # Core
    r3 = int(r * 0.3)
    draw.ellipse([cx - r3, cy - r3, cx + r3, cy + r3], fill=COLORS[2])

    out = os.path.join(ICONS_DIR, f"icon{size}.png")
    img.save(out, "PNG")
    print(f"Wrote {out}")

print("Done.")

File: generate-icons.mjs
// Generate PNG icons from the amber design.
//
// Run from inside the project directory:
//   node generate-icons.mjs
// Requires the `canvas` npm module to be installed in a reachable node_modules.

import { createCanvas } from 'canvas';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const iconsDir = path.join(__dirname, 'icons');

// Ensure target directory exists before writing (first-run safety).
fs.mkdirSync(iconsDir, { recursive: true });

const sizes = [16, 32, 48, 128];

for (const size of sizes) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // Amber circle
  ctx.clearRect(0, 0, size, size);
  const r = size * 0.42;
  const cx = size / 2, cy = size / 2;

  // Outer ring
  ctx.beginPath();
  ctx.arc(cx, cy, r, 0, Math.PI * 2);
  ctx.fillStyle = '#d4a017';
  ctx.fill();

  // Inner glow
  ctx.beginPath();
  ctx.arc(cx, cy, r * 0.7, 0, Math.PI * 2);
  ctx.fillStyle = '#f0c040';
  ctx.fill();

  // Core
  ctx.beginPath();
  ctx.arc(cx, cy, r * 0.3, 0, Math.PI * 2);
  ctx.fillStyle = '#b8920f';
  ctx.fill();

  const buf = canvas.toBuffer('image/png');
  const outPath = path.join(iconsDir, `icon${size}.png`);
  fs.writeFileSync(outPath, buf);
  console.log(`Wrote ${outPath} (${buf.length} bytes)`);
}

console.log('Done.');
